core/src/main/java/org/bitcoinj/core/LockTime.java

```diff
@@ -99,26 +135,24 @@ public long rawValue() {
      * @return true if lock time is set
      */
     public boolean isSet() {
-        return value > 0;
+        return !(this instanceof NoLock);
     }
 
     /**
      * Determine if this lock time is specified as a block height. That means its raw value is below {@link #THRESHOLD}.
+     * TODO: this maps NoLock into a BlockLock. I'm not sure this is what we want, but I think some use-cases need this.
      * @return true if specified as a block height
      */
     public boolean isBlockHeight() {
-        return value < THRESHOLD;
+        return this instanceof HeightLock || this instanceof NoLock;
     }
 
     /**
      * Gets the lock time as a block height.
      * @return lock time as a block height
-     * @throws IllegalStateException if the lock time is not specified as a block height
      */
-    public int blockHeight() {
-        if (!isBlockHeight())
-            throw new IllegalStateException("lock time not specified as a block height");
-        return Math.toIntExact(value);
+    public OptionalInt getBlockHeight() {
```

Comment:
`blockHeight()`?

Reply:
We don't want this to conflict with the `.blockHeight()` method in the subclass. That's why I added the `get`...

Reply:
... and of course we could remove this altogether.

Reply:
Removed.